### ISSUE TITLE
[SEP] token

### ISSUE THREAD

**Issue by MichalPitr** [donggyukimc/Inverse-cloze-task#2]
Hi, 
Is there a reason you do not add a `[SEP] ` token at the end of a sentence/context? 

Thanks a lot
Michal

**Comment by donggyukimc** [comment]
Nope, there is no particular reason for that in the source code of this repo, lol.

As you know, many works using fine-tuning on LM utilize the inputs with the form of [CLS] + tokens + [SEP], even though they do not use them both.

However, when you are trying to do dense retrieval task, you can try various approaches to make your semantic vectors (text embedding). 

For example, you can try to make  embedding vectors consisting of vectors of [CLS] and [SEP] (concat). 

This approach gave me interesting results. 

But, results will be different according to your circumstance.

**Comment by MichalPitr** [comment]
@donggyukimc Thanks for the reply!

Makes sense, I've done some tests with the default pooling and using just the [CLS] embedding and it seems using the CLS embedding has the highest precision@k (at least given the limited testing I've done).

My initial concern was relevant to the fact that bert models were trained with [SEP], but since ICT usually involves a lot of data (Wikipedia in my case), the model simply learns to work without it. Would be interesting to see whether the absence of [SEP] has some effect on the rate of change of the loss function early on in the training. May run some tests on this just for the lols.

Again thanks for the clarification!
